src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/decisionmaker/deciders/Publisher.java

```diff
@@ -0,0 +1,64 @@
+package com.amazon.opendistro.elasticsearch.performanceanalyzer.decisionmaker.deciders;
+
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.decisionmaker.actions.Action;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.grpc.FlowUnitMessage;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.framework.core.GenericFlowUnit;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.framework.core.NonLeafNode;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.scheduler.FlowUnitOperationArgWrapper;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+public class Publisher extends NonLeafNode<GenericFlowUnit> {
+
+    private static final Logger LOG = LogManager.getLogger(Publisher.class);
+
+    private Collator collator;
+    private boolean isMuted = false;
+
+    public Publisher(int evalIntervalSeconds, Collator collator) {
+        super(0, evalIntervalSeconds);
+        this.collator = collator;
+    }
+
+    @Override
+    public GenericFlowUnit operate() {
```

Comment:
maybe we can create a dedicated EmptyFlowUnit for the top level vertex in graph ?

Reply:
I'm of split opinion here, but created one so as to explicitly identify the terminal node.